packages/api/src/controllers/stream.js

```diff
@@ -53,6 +54,51 @@ const hackMistSettings = (req, profiles) => {
   });
 };
 
+async function validatePushTarget(userId, profileNames, pushTargetRef) {
+  const { profile, id, spec } = pushTargetRef;
+  if (!profileNames.has(profile) && profile !== "source") {
+    throw new BadRequestError(
+      `push target must reference existing profile. not found: "${profile}"`
+    );
+  }
+  if (!!spec === !!id) {
```

Comment:
Maybe a smidge more explicit:
```suggestion
  if (!spec && !id) {
```

Reply:
The difference with that one would be that we wouldn't catch when **both** are set, which IMO should also be an error here. I added a test case for that scenario which would break if we changed the `===` to a `&&` here: https://github.com/livepeer/livepeer-com/blob/174374a5e87fb0a280f0bbb176b2237e0a06f71f/packages/api/src/controllers/stream.test.js#L255

I can actually remove one of the `!` and have the same desired behavior though, WDYT?
```suggestion
  if (!spec === !id) {
```

Otherwise, do you think we should just ignore the spec in case both fields are set, only using the ID?

Reply:
Ah, good catch, I didn't think of the "both" case. You can leave as-is